src/main/java/it/unisa/Amigo/repository/controller/RepositoryController.java

```diff
@@ -16,36 +18,97 @@
 import org.springframework.web.multipart.MultipartFile;
 
 import java.util.List;
+import java.util.Set;
 
 
 @Controller
 public class RepositoryController {
     @Autowired
-    private DocumentoService documentoService;
+    private RepositoryService repositoryService;
+
+    @Autowired
+    private GruppoService gruppoService;
+
+    /**
+     * Permette di verificare se l'utente il Responsabile del PQA.
+     *
+     * @return true se l'utente è il Responsabile del PQA.
+     */
+
+    private boolean isResponsabilePQA() {
+        Set<Role> ruoli = gruppoService.getAuthenticatedUser().getUser().getRoles();
```

Comment:
CODE SMELL ALERT! Si può fare un metodo per ottenere direttamente i ruoli dell'utente loggato in GruppoService